sql/parser/encode.go

```diff
@@ -38,16 +38,37 @@ var (
 )
 
 func encodeSQLString(buf []byte, in []byte) []byte {
+	// Changed if escape sequences necessary. See:
+	// http://www.postgresql.org/docs/9.0/static/sql-syntax-lexical.html
+	var strMod byte
+	for _, ch := range in {
+		if encodedChar := encodeMap[ch]; encodedChar != dontEscape {
+			// If a character which we want to escape is found, update strMod.
+			strMod = 'e' // TODO(tschottdorf) should this be 'E'?
+			break
+		}
+	}
```

Comment:
I've backed out the part of the change that avoided encoding `'` and `\` where possible from this method and rewrote this method. Looks pretty sleek now and the only thing that has changed is that `'` and `\` lead to an escaped string.